app/submitter/converter_v2.py

```diff
@@ -134,4 +136,22 @@ def get_payload_data(
             "lists": list_store.serialize(),
         }
 
+        if answer_codes := schema.json.get("answer_codes"):
+            get_filtered_answer_codes(answer_codes, data)
```

Comment:
- The function `get_filtered_answer_codes` implies you get sometime back but it is mutating which is always good to avoid. I would update this to return the answer codes then assign `data["answer_codes"] = filtered_answer_codes` here. 

- It's input should just be, `answer_codes` and `answer_ids_to_keep`. It should only be passed what it needs.

- Answer codes should only be assigned if we have at least one. So empty answer codes (because we have no answers) shouldn't be assigned. We can look at changing "answers" and "lists" to match in future as no need to add keys that are empty. But currently validation for answer_codes enforce, it must have at least one item.

Reply:
Updated the method now 👍